src/BinDeps.jl

```diff
@@ -1,37 +1,32 @@
 module BinDeps
     importall Base
 
+    using Compat
+
     export @make_run, @build_steps, find_library, download_cmd, unpack_cmd,
             Choice, Choices, CCompile, FileDownloader, FileRule,
             ChangeDirectory, FileDownloader, FileUnpacker, prepare_src,
             autotools_install, CreateDirectory, MakeTargets, SystemLibInstall
 
-    if VERSION >= v"0.4.0-dev+3844"
-        import Base.Libdl: dlext, dlpath, RTLD_LAZY, DL_LOAD_PATH
-    else
-        const dlext = isdefined(Base.Sys, :shlib_ext) ? Base.Sys.shlib_ext : Base.Sys.dlext # Julia 0.2/0.3 compatibility
```

Comment:
Actually, `dlpath` was not implemented in any form until 0.3, it looks like.  So supporting 0.2 is hopeless (the `dlpath` function uses a C implementation, `jl_pathname_for_handle`, that was not added until 0.3).